src/system/TLVPacketBufferBackingStore.cpp

```diff
@@ -31,7 +31,7 @@ namespace System {
 CHIP_ERROR TLVPacketBufferBackingStore::OnInit(chip::TLV::TLVReader & reader, const uint8_t *& bufStart, uint32_t & bufLen)
 {
     bufStart = mHeadBuffer->Start();
-    bufLen   = mHeadBuffer->DataLength();
+    bufLen   = static_cast<uint32_t>(mHeadBuffer->DataLength());
```

Comment:
Needs to be checking the cast is safe, erroring if not?

This is all going to come back and bite us sometime.  Please fix all such callsites.